java/query/src/test/java/sleeper/query/lambda/SqsQueryProcessorLambdaIT.java

```diff
@@ -330,13 +336,22 @@ public void shouldSetStatusOfQueryToCOMPLETEDWhenOnlyOneSubQueryIsCreated() thro
                 .queryId("abc")
                 .regions(List.of(new Region(List.of(range1, range2, range3))))
                 .build();
+
         processQuery(query);
+        processLeafPartitionQuery();
 
         // Then
         assertThat(queryTracker.getAllQueries())
                 .usingRecursiveFieldByFieldElementComparatorIgnoringFields("lastUpdateTime", "expiryDate")
-                .containsExactly(trackedQuery()
+                .containsExactlyInAnyOrder(trackedQuery()
+                        .queryId("abc")
+                        .subQueryId("-")
+                        .lastKnownState(COMPLETED)
+                        .recordCount(10L)
+                        .build(),
+                        trackedQuery()
                         .queryId("abc")
+                        .subQueryId(queryTracker.getAllQueries().get(1).getSubQueryId())
```

Comment:
I don't think this assertion will pass reliably as there's no guarantee of the ordering in the scan results from getAllQueries. There are a few options:

1. Find the first sub query ID in the list that isn't "-"
2. Ignore the sub query ID in the assertion, as in the other tests
3. Inject some code to generate the sub query IDs so you can fix specific values for them - this would need to happen in QueryExecutor

Any of these would work.

Reply:
Updated to option 1